Src/BootCamp.Chapter/TextTable.cs

```diff
@@ -30,11 +32,141 @@ public static class TextTable
+                {
+                    endPos = i - 1;
+                    messageLine[messageCount++] = message.Substring(beginPos, endPos - beginPos + 1);
+                    beginPos = i + 2;
+                }
+            }
+
+            if (numOfMessageLines == 0 && message.Length > 0)
+            {
+                messageLine[0] = message;
+            }
+
+            if (endPos <= message.Length)
+            {
+                messageLine[messageCount++] = message.Substring(beginPos, message.Length - beginPos);
+            }
+
+            return messageLine;
+        }
+
+        private static int GetMaxMessageLineLength(string[] messageLine)
+        {
+            var maxLength = 0;
+
+            for (var i = 0; i < messageLine.Length; i++)
+            {
+                if (maxLength < messageLine[i].Length)
+                {
+                    maxLength = messageLine[i].Length;
+                }
+            }
+
+            return maxLength;
+        }
+
+        private static string BuildTable(string[] messageLine, int maxLength, int padding)
+        {
+            var tableLine = BuildBorderLine(maxLength, padding, "+", "-");
+            var table = tableLine + Environment.NewLine;
+
+            for (var i = 0; i < padding; i++)
+            {
+                tableLine = BuildBorderLine(maxLength, padding, "|", " ");
+                table += tableLine + Environment.NewLine;
+            }
+
+            for (var i = 0; i < messageLine.Length; i++)
+            {
+                tableLine = BuildMessageLine(maxLength, padding, "|", messageLine[i]);              
+                table += tableLine + Environment.NewLine;
+            }
+
+            for (var i = 0; i < padding; i++)
+            {
+                tableLine = BuildBorderLine(maxLength, padding, "|", " ");
+                table += tableLine + Environment.NewLine;
+            }
+
+            tableLine = BuildBorderLine(maxLength, padding, "+", "-");
+            table += tableLine + Environment.NewLine;
+
+            return table;
+        }
+
+        private static string BuildBorderLine(int maxLength, int padding, string sideBorder, string edgeBorder)
+        {
+            var line = sideBorder;
+            for (var i = 0; i < maxLength + 2 * padding; i++)
+            {
+                line += edgeBorder;
+            }
+            line += sideBorder;
+
+            return line;
+        }
+
+        private static string BuildMessageLine(int maxLength, int padding, string sideBorder, string messageLine)
+        {
+            var line = sideBorder;
+            for (var i = 0; i < (maxLength - messageLine.Length) / 2 + padding; i++)
+            {
+                line += " ";
+            }
+
+            line += messageLine;
+
+            for (var i = 0; i < (maxLength - messageLine.Length) / 2 + padding + ((maxLength - messageLine.Length) % 2); i++)
+            {
+                line += " ";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4955013)